### ISSUE TITLE
Include MFA modules in this recipe

### ISSUE THREAD

**Issue by brynwhyman** [silverstripe/cwp#258]
## Overview

With the next minor release of CWP, the multi-factor authentication module suite should be included by default for all sites upgrading. 

### What's in the MFA module suite?
- [Silverstripe MFA](https://github.com/silverstripe/silverstripe-mfa)
- [TOTP](https://github.com/silverstripe/silverstripe-totp-authenticator)
- [Security token, via WebAuthn](https://github.com/silverstripe/silverstripe-webauthn-authenticator)

All of these should be included by default, with some configuration available for development agencies to tweak the requirements based on their guidance or discussions that are had with Site Owners.

## Release notes (this is supported by a number of internal tasks by the CWP team):
- Remember, enforcing MFA is still left with CMS admin
- Clear documentation in the release change logs will prompt Developers doing the upgrade to have a conversation with the Site Owner that they'll get MFA at the same time. 
- Ensuring a developer is included in testing MFA with a site is important.

The MFA module bundles the [login-forms module](https://github.com/silverstripe/silverstripe-login-forms) which will override any existing login page (Security.ss) template. This could be a detail that warrants the MFA functionality being installed but embargoed until further approval - but I'd expect this is something a development agency should be involved in and can make happen.

If for whatever reason the dev agency doesn't think MFA is suitable for a site on upgrade, [there is the opportunity to disable the login redirect to MFA via `yaml`](https://github.com/silverstripe/silverstripe-mfa/blob/master/docs/en/local-development.md#local-development), without having to alter their `composer.json` file.

**Comment by chillu** [comment]
@brynwhyman Is this a duplicate of the [Epic: MFA is in CWP by default](https://github.com/silverstripeltd/product-issues/issues/153)?

**Comment by brynwhyman** [comment]
> Is this a duplicate of the Epic: MFA is in CWP by default?

@chillu no, https://github.com/silverstripeltd/product-issues/issues/153 was created as an internal epic to capture all related work under one card.

**Comment by brynwhyman** [comment]
I'd like to suggest that we leave out the [Security token](https://github.com/silverstripe/silverstripe-webauthn-authenticator) option as what's included in the CWP module by default. There's a number of edge cases and potential for havoc without the site owner and development agency taking the time to understand these before requesting this option, including:
- Security key cannot be used over different domains connected to the same CMS (accessing a subsite `/admin` URL over a different top-level domain)
- Security key needs to be reset for all users following a snapshot restore between different Silverstripe platform environments (where the top-level domain has changed). E.g. promoting UAT content to Prod.
- Security key needs to be reset for all users following a general domain change for the site (company-name.com to new-company-name.com)

There's another open issue suggesting to disable the security token functionality if the subsites module is installed, but that's not covering all the edge cases. I believe something like this should still happen, but it shouldn't block sites getting access to the much more popular and accessible MFA method, TOTP.

**Comment by brynwhyman** [comment]
@Cheddam has noted that there's at least one issue in the login-forms module that we'll want to look at as part of this issue. Will add to our internal epic.

**Comment by brynwhyman** [comment]
We're talking about a few different options to actually make this happen. Should we:
- add it to cwp-recipe-core
- add it to cwp-recipe-cms
- add it to cwp-installer (the upgrade instructions that's provided in each release references the modules in the cwp-installer). Noting that this will probably only serve new site builds

Or:
- We keep it as effectively 'opt-in' as it's own module set and push it hard in the release documentation.

**Comment by chillu** [comment]
Bryn mentioned internally that the current inclination is to leave it out of the recipes, which would introduce `silverstripe/mfa` and `silverstripe/login-forms` on minor upgrades. Reasons:

 * While MFA can stay disabled by default, `login-forms` will be active. Note that we've added `login-forms` in `silverstripe/installer` (outside of cwp) rather than the recipes, so it won't be automatically included on upgrades there either.
 * With `login-forms` active by default, it'll override any custom login form styling. That's kind of the point of this module (and necessitated by the complex styling required during the MFA flows), but also might not be desired by customers
 * `login-forms` might also mess with any custom SSO flows
 * `mfa` is not compatible with LDAP, RealMe or SAML based logins

So in summary, I recommend that we use the third option described here:

> add it to cwp-installer (the upgrade instructions that's provided in each release references the modules in the cwp-installer). Noting that this will probably only serve new site builds

Also, the installer would only include the TOTP MFA method, not the WebAuthn method since that requires a bit more thought by developers (subsites, multi env usage). This should be outlined in the upgrade docs though, to ensure that WebAuthn is actually considered. It's far more secure than TOTP (lower phishing potential, domain verification built in)

**Comment by chillu** [comment]
I'm also advocating for the same approach in core: Add this to the installer rather than recipes. https://github.com/silverstripe/silverstripe-installer/issues/280

**Comment by brynwhyman** [comment]
Thanks for the comments @chillu, I agree that pushing it through the CWP-installer is still a good thing to do.

Do you have any ideas on how to ensure that Developers get the TOTP encryption key .env variable? It's on the CWP platform by default so a deployed site would be covered, but can we only rely on documentation to ensure Developers are aware of this?
